nh_odoo_fixes/models/datetime_utils.py

```diff
@@ -0,0 +1,69 @@
+# -*- coding: utf-8 -*-
+"""Contains various useful methods for managing datetimes."""
+from datetime import datetime
+
+from openerp import models
+from openerp.tools import DEFAULT_SERVER_DATETIME_FORMAT as DTF
+
+
+class DatetimeUtils(models.AbstractModel):
+
+    _name = 'datetime_utils'
+
+    @classmethod
+    def zero_microseconds(cls, date_time):
+        """
+        Return the passed date_time with any microseconds set to 0.
+
+        :param date_time:
+        :type date_time: datetime or str
+        :return:
+        """
+        if isinstance(date_time, str):
+            date_time_split = date_time.split('.')
+            if len(date_time_split) > 2:
+                raise ValueError(
+                    "Datetime contains more than one period and so is not in "
+                    "the expected format. Method will not work successfully "
+                    "for other formats in its current state."
+                )
+            return date_time_split[0]  # Omit microseconds after period.
+        return date_time.replace(microsecond=0)
+
+    @classmethod
+    def zero_seconds(cls, date_time):
+        """
+        Return the passed date_time with any seconds and microseconds set to 0.
+
+        :param date_time:
+        :type date_time: datetime
+        :return:
+        """
+        return date_time.replace(second=0, microsecond=0)
+
+    @classmethod
+    def reformat_server_datetime_for_frontend(cls, date_time,
+                                              date_first=False):
+        """
+        Reformat a datetime in Odoo's 'default server datetime format'
+        (see imports) to one more appropriate for the front end.
+
+        Can choose whether the date or time comes first.
+
+        :param date_time:
+        :type date_time: str
+        :param date_first:
+        :type date_first: bool
+        :return:
+        :rtype: str
+        """
+        date_time = cls.zero_microseconds(date_time)
+        date_time = datetime.strptime(date_time, DTF)
+        date = '%d/%m/%Y'
+        time = '%H:%M'
+        if date_first:
```

Comment:
It has some benefits as it converts the object passed to a string so if you pass the actual date & time objects instead of the string representation of those objects it'll display them correctly instead of blowing up, same with None objects for instance

it's also more performant http://stackoverflow.com/a/16514440/397801